Client/Assets/Favicons.js

```diff
@@ -0,0 +1,32 @@
+(function() {
+
+ if (window.__firefox__ === undefined) {
+    window.__firefox__ = { }
+ }
+ 
+ function Favicons() {
+ }
+
+ Favicons.prototype = {
+    selectors: ["link[rel~='icon']",
+                "link[rel='apple-touch-icon']",
+                "link[rel='apple-touch-icon-precomposed']"],
+    getAll: function() {
+        var res = []
+        for (var s = 0; s < this.selectors.length; s++) {
+            var icons = document.querySelectorAll(this.selectors[s])
+            for (var i = 0; i < icons.length; i++) {
+                res.push(icons[i].href);
+            }
+        }
+
+        if (res.length == 0) {
+            res.push(document.location.origin + "/favicon.ico")
+        }
+
+        return res;
+    }
+ }
+
+ window.__firefox__.Favicons = new Favicons()
+})()
```

Comment:
This is nice. Just a note for myself that I need to restructure ReaderMode.js to use a similar setup. The anonymous function trick is nice. https://bugzilla.mozilla.org/show_bug.cgi?id=1127853